flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/optimize/StreamNonDeterministicUpdatePlanVisitor.java

```diff
@@ -200,7 +201,8 @@ public StreamPhysicalRel visit(
                 || rel instanceof StreamPhysicalSortLimit
                 || rel instanceof StreamPhysicalTemporalSort
                 || rel instanceof StreamPhysicalWatermarkAssigner
-                || rel instanceof StreamPhysicalExchange) {
+                || rel instanceof StreamPhysicalExchange
+                || rel instanceof StreamPhysicalDeltaJoin) {
```

Comment:
Actually it should same with `CommonPhysicalJoin`... But at first we don't need to support delta join to resolve NDU and this is why I'm not adding it in `StreamNonDeterministicUpdatePlanVisitor` in the previous pr.